reversed comparison logic of length check within RedstoneAware:getOutput()

SanAndreaP opened the pull request:
fixes #3439

SanAndreaP committed:
reversed comparison logic of RedstoneAwares' getOutput() length check, fixes #3439

v1993 commented:
Can we *please* get this one merged? Broken redstone is a huge problem.

Ocawesome101 commented:
Seconded.

CaitlynMainer commented:
@Vexatos Are you able to merge PRs?  If not @fnuecke / @payonel if I could drag one of you out of retirement to assign someone permissions to be able to review and merge PRs that'd be appreciated! <3

fnuecke commented on src/main/scala/li/cil/oc/common/tileentity/traits/RedstoneAware.scala:
```diff
@@ -80,7 +80,7 @@ trait RedstoneAware extends RotationAware {
   def getOutput: Array[Int] = EnumFacing.values.map{ side: EnumFacing => _output(toLocal(side).ordinal) }
 
-  def getOutput(side: EnumFacing) = if (_output != null && _output.length <= toLocal(side).ordinal())
+  def getOutput(side: EnumFacing) = if (_output != null && _output.length >= toLocal(side).ordinal())
```
Shouldn't this be `_output.length > toLocal(side).ordinal()` then?
Since `toLocal(side).ordinal()` is used as the index, so if it's `== _output.length` it'd be out of bounds?

SanAndreaP replied:
Yes, you're right.

SanAndreaP committed:
Update RedstoneAware.scala

comparison should be checking if array length is bigger than and *not* equal to the ordinal

fnuecke commented:
Thanks for the fix :)
